src/components/Map.js

```diff
@@ -0,0 +1,62 @@
+import GoogleMap from 'google-map-react'
+import PropTypes from 'prop-types'
+import React from 'react'
+
+import Cluster from './Cluster'
+import Location from './Location'
+
+const Map = (props) => {
+  const {
+    bootstrapURLKeys,
+    view,
+    markerData,
+    showLocations,
+    onZoomAnimationEnd,
+    onDragEnd,
+    onClusterClick,
+    onLocationClick,
+  } = props
+
+  return (
+    <GoogleMap
+      bootstrapURLKeys={bootstrapURLKeys}
+      center={view.center}
+      zoom={view.zoom}
+      onZoomAnimationEnd={onZoomAnimationEnd}
+      onDragEnd={onDragEnd}
+    >
+      {markerData.map((marker, index) =>
+        showLocations ? (
+          <Location
+            key={marker.id}
+            onClick={onLocationClick}
+            id={marker.id}
+            lat={marker.lat}
+            lng={marker.lng}
+          />
+        ) : (
+          <Cluster
+            key={index}
+            onClick={onClusterClick}
+            lat={marker.lat}
+            lng={marker.lng}
+            count={marker.count}
+          />
+        ),
+      )}
+    </GoogleMap>
+  )
+}
+
+Map.propTypes = {
+  bootstrapURLKeys: PropTypes.object.isRequired,
+  view: PropTypes.object.isRequired,
+  markerData: PropTypes.arrayOf(PropTypes.object).isRequired,
+  showLocations: PropTypes.bool.isRequired,
```

Comment:
It's a pretty interesting approach to take a list and a bool to represent one of [array of locations, array of clusters].

However, there are some potential downsides to this approach:

- Your location object and cluster object have a different schema. Cluster has a count property, and location has an id. In that sense, typing doesn't make sense here, because it's an array of two different types.
- Potentially, we may want to show both clusters and locations on the map. This is not possible with the current approach.
- Since you made parallel callbacks `onClusterClick`, `onLocationClick`, it would also make sense to make parallel array props for those. OTOH, you could make one callback, `onMarkerClick(details, isLocation)` but that doesn't seem clean to me.

In general it boils down to the fact that Cluster and Location are different types.